File: Backend/Nodejs/src/routes/album.ts
import { Router } from "express";
import multer from "multer";
import { s3Storage } from "../libs/s3client";
import { AlbumController } from "../controllers/album";

export const albumRouter = Router();
const uploadAlbum = multer({storage: s3Storage({userType:"album"})})

albumRouter.get("/",AlbumController.getAlbums);
albumRouter.get("/:id",AlbumController.getAlbum);
albumRouter.get("/:id/songs",AlbumController.getSongs);
albumRouter.post("/newAlbum",uploadAlbum.single("cover"),AlbumController.createAlbum);
albumRouter.post("/addSong",AlbumController.addSong);
albumRouter.post("/removeSong",AlbumController.removeSong);
albumRouter.patch("/",uploadAlbum.single("cover"),AlbumController.editAlbum);
albumRouter.delete("/:id",AlbumController.deleteAlbum);

File: Backend/Python/src/routes/album.py
from flask import Blueprint
from src.controllers.album import AlbumController

album_blueprint = Blueprint("blueprint", __name__)

album_blueprint.route("/", methods=["GET"])(AlbumController.get_albums)
album_blueprint.route("/<int:id>/", methods=["GET"])(AlbumController.get_album)
album_blueprint.route("/<int:id>/songs/", methods=["GET"])(AlbumController.get_songs)
album_blueprint.route("/newAlbum", methods=["POST"])(AlbumController.create_album)
album_blueprint.route("/addSong", methods=["POST"])(AlbumController.add_song)
album_blueprint.route("/removeSong", methods=["POST"])(AlbumController.remove_song)
album_blueprint.route("/<int:id>", methods=["DELETE"])(AlbumController.delete_album)
album_blueprint.route("/", methods=["PATCH"])(AlbumController.edit_album)
